src/TLuaInterpreter.cpp

```diff
@@ -10147,6 +10147,18 @@ int TLuaInterpreter::getTime(lua_State* L)
 }
 
 
+// syntax: getEpoch()
+// returns: seconds since unix epoch with milliseconds (e.g. 1523555867.191)
+int TLuaInterpreter::getEpoch(lua_State *L)
+{
+    QDateTime time = QDateTime::currentDateTime();
```

Comment:
Does this return the datetime you think it does? The documentation says:
> Returns the current datetime, as reported by the system clock, in the local time zone.

It also remarks:
> `QDateTime` uses the system's time zone information to determine the offset of local time from UTC. If the system is not configured correctly or not up-to-date, QDateTime will give wrong results as well.

ℹ️ Whilst \*nix systems are pretty good with keeping track of time in UTC (thank `ntpd`) and then presenting users with the correct time in *their* locale (*Wallclock* time) - Windows is not so hot when DST changes take effect IIRC. {Mind you even ntpd gets a bit funky around leap-seconds and there has been 26 of them I think since the Epoch}  If a package/module is timing things to other users on other systems in other TimeZones it does help if everyone has the same idea of what the time is...!

I hope we test this properly on Windows because I think that one is the most likely to present odd results.

At the third stroke the time - according to my ntpd-running Raspberry Pi displaying the seconds since epoch on its 2x16 character LCD display will be: ... beep ... beep ... beep ... **1523583930**

Reply:
Correct. The intention of this addition is to have a more performant way to get current epoch time plus milliseconds. This function cannot assure the system clock is accurate nor is it partial to timezone. My personal use case for this is I have a script that keeps a log of multiple events from the MUD along with their timestamps (need milliseconds too). Doing format conversion or using the stopwatch functions were a workaround / not ideal.

Reply:
Yeah, the stopwatches are limited to 24 hours anyhow.

OK, I guess this comment is a ***no action***.